bmm/src/main/java/org/openehr/bmm/v2/persistence/PBmmSingleProperty.java

```diff
@@ -29,14 +38,29 @@ public void setType(String type) {
     @Override
     @JsonIgnore
     public PBmmSimpleType getTypeRef() {
-        if(serializedTypeRef != null) {
+        if (serializedTypeRef != null) {
             return serializedTypeRef;
-        } else if(getTypeDef() == null) {
+        } else if (getTypeDef() == null) {
             return new PBmmSimpleType(type);
         }
         return getTypeDef();
     }
 
+    @Override
+    public BmmUnitaryProperty createBmmProperty(BmmClassProcessor classProcessor, BmmClass bmmClass) {
+        PBmmSimpleType typeRef = getTypeRef();
+        if (typeRef != null) {
+            BmmSimpleType bmmType = typeRef.createBmmType(classProcessor, bmmClass);
+            if (bmmType != null) {
+                BmmUnitaryProperty bmmProperty = new BmmUnitaryProperty(getName(), bmmType, getDocumentation(), nullToFalse(isMandatory()), nullToFalse(isComputed()));
+                populateImBooleans(bmmProperty);
+                return bmmProperty;
+            }
```

Comment:
`createBmmType` always returns a non-null result. This if statement can be removed.